pkg/errors/resource_reference_error.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	// ResourceReferenceOrIDRequired indicates that the user failed to specify
+	// either the actual referenced identifier or an AWSResourceReferenceWrapper
+	ResourceReferenceOrIDRequired = fmt.Errorf("resource reference wrapper" +
+		"or ID required")
```

Comment:
Above, there should be a space between "wrapper" and "or". I find it easier to write these kinds of  multi-line strings like this, fwiw:

```go
	ResourceReferenceOrIDRequired = fmt.Errorf(
		"resource reference wrapper or ID required",
	)
```

for longer strings you can do:

```go
	ResourceReferenceOrIDRequired = fmt.Errorf(
		"resource reference wrapper or ID required"+
			" my continuation line goes here"+
			" along with yet another long line",
	)
```

which makes it a bit easier to detect things like missing spaces, since you put all the spaces at the beginning of each continuation line of text.